### PR TITLE
Implementing suggestions 2: Surge bubble particles, configuration, Extraction iron golem viability

### PR DESCRIPTION
**Pull request by PrideInt**
### Implementing suggestions from Discord: Part Two.

> Part One here: https://github.com/ProjectKorra/ProjectKorra/pull/1329

## Additions

- Surge bubble particles. ***Suggested by member: `.blackxlight`***
    - When `SurgeWave` or `SurgeWall` interacts with water blocks (e.g., if used underwater), bubble particles are displayed--similarly to `Torrent` and `WaterManipulation`.
- `MetalClips`' shoot speed and cooldown can be configurable now. ***Suggested by member: `.blackxlight`***
- `SurgeWall` select range configurability. ***Suggested by member: `prusakovaleksandr`***
- `Extraction` viability on **IRON GOLEMS**. ***Suggested by member: `.blackxlight`***
   - `Extraction` drops **IRON NUGGETS** when used on **IRON GOLEMS**. Players may configure the amount dropped in the config.

> [!NOTE]
> *Will manually resolve merge conflicts from either pull request when one is merged into **master**.*

### CODE REVIEW

**Comment by PrideInt** [comment]
### Up next:

- "Revert time for Firebending's grief (with `FireBlastCharged` and `Combustion`), just like Earth and Water". ***Suggested by member: `.blackxlight`***
- Moon cycle night factor: ***Suggested by member: `legendary_ocean`***
    - "*Set NightFactor's config to be valid for moon phases. For example, a NightFactor value in the config is 1.5 (+%50) and if there is a New moon at night, %20 of this value will be taken. So the result will be +%10. If there is a Gibbous moon, %80 of this value will be taken and this time the result will be +%40*"

    mooncyclefactor
- Extraction viability on **IRON GOLEMS**. ***Suggested by member: `.blackxlight`***

**Comment by PrideInt** [comment]
### Soon/separate pull request:

- Bloodbending weight factor. ***Suggested by member: `.blackxlight`***
   - "*Configurable. "weight" mechanic for bloodbending: allow us to assign "weights" to certain mobs (more blood to bend = harder to bloodbend). Mobs with higher weights will be harder to bloodbend (move around), and their throw distance will be reduced. Vice versa for mobs with lighter weights.*"

File: core/src/com/projectkorra/projectkorra/earthbending/metal/Extraction.java [hunk 1]
```diff
@@ -39,6 +46,12 @@ public Extraction(final Player player) {
 		this.tripleChance = getConfig().getDouble("Abilities.Earth.Extraction.TripleLootChance");
 		this.cooldown = getConfig().getLong("Abilities.Earth.Extraction.Cooldown");
 		this.selectRange = getConfig().getInt("Abilities.Earth.Extraction.SelectRange");
+		this.ironGolemDrops = getConfig().getInt("Abilities.Earth.Extraction.IronGolem.Drops");
+		this.ironGolemDamage = getConfig().getDouble("Abilities.Earth.Extraction.IronGolem.Damage");
+		this.ironGolemDropMaterial = Material.getMaterial(getConfig().getString("Abilities.Earth.Extraction.IronGolem.DropMaterial"));
+		if (this.ironGolemDropMaterial == null) {
+			this.ironGolemDropMaterial = Material.IRON_INGOT;
+		}
```
Comment (JustAHuman-xD) [inline core/src/com/projectkorra/projectkorra/earthbending/metal/Extraction.java:54]: is there a reason this doesn't default to the config default option? (Iron nugget)

Comment (PrideInt) [reply core/src/com/projectkorra/projectkorra/earthbending/metal/Extraction.java:54]: No reason. You are right, though. 

However, I feel that making this configurable makes no sense. Perhaps we could make the drop item from IRON GOLEMS as something like a Bernoulli, where the probability/chance of dropping ingots `p` (IRON INGOT) is defined in the config (like 40; 40%) and the probability of dropping nuggets would be `100 - p` (60% here). It gives users the freedom to also choose exactly what material it drops by setting the chance to either 0 (always nuggets) or 100 (always ingots).

Comment (PrideInt) [reply core/src/com/projectkorra/projectkorra/earthbending/metal/Extraction.java]: - Addressed in: https://github.com/ProjectKorra/ProjectKorra/pull/1333/commits/579f52d0a7535946e16b23fcf29abcfbccfb6916
   - `Extraction` on IRON GOLEMS will now only drop iron nuggets.

**Comment by PrideInt** [comment]
> ### Up next:
> * "Revert time for Firebending's grief (with `FireBlastCharged` and `Combustion`), just like Earth and Water". _**Suggested by member: `.blackxlight`**_
> * ~`SurgeWall` select range configurability. _**Suggested by member: `prusakovaleksandr`**_~
> * ~Extraction viability on **IRON GOLEMS**. _**Suggested by member: `.blackxlight`**_~

Moving Firebending grief reversion to a whole other pull request. Physics updating from fire griefing is a more complicated issue than previously thought and would require some changes.

**Comment by PrideInt** [comment]
@ChristopherWMM this PR should be ready to go.

**Review by JustAHuman-xD — APPROVED** [review]
LGTM

**Review by ChristopherWMM — APPROVED** [review]

**Pull request merged**